proposals/0355-regex-syntax-run-time-construction.md

```diff
@@ -94,6 +94,8 @@ extension Regex where Output == AnyRegexOutput {
 }
 ```
 
+These initializers treat multi-line patterns the same way as [multi-line regex literals][multi-line-regex]. If a pattern spans multiple lines, [extended syntax mode](#extended-syntax-modes) `(?x)` is enabled by default. This means that whitespace in the pattern becomes non-semantic, and end-of-line comments are supported with `# comment` syntax. This only affects literal newline characters in the pattern, the `\n` escape sequence is unaffected. API to customize this behavior is deferred as future work.
```

Comment:
How does this relate to the behavior of other engines we're citing? What do they do for regexes that have those byte sequences inside them?

For the literal, I thought it also has to begin with `\s*?\n`.

If there is any deviation for literals from run-time strings, I'd expect the run-time strings to maximize compatibility. I don't want to create a large or formal dialect for the literal, but I could see some minor differences being useful as they relate to language integration.

Reply:
Interesting point! I think this probably deserves more discussion. If we did require a leading newline that would mean that:

```swift
let pattern = """
 (a | b | c)
 [ d e f ]+
"""
```

would not use extended syntax, but:


```swift
let pattern = """

 (a | b | c)
 [ d e f ]+
"""
```

would. IMO this might be surprising, what do you think?